core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -154,15 +164,24 @@ final class ShuffleBlockFetcherIterator(
     while (iter.hasNext) {
       val result = iter.next()
       result match {
-        case SuccessFetchResult(_, address, _, buf, _) =>
+        case SuccessFetchResult(_, address, size, buf, _) =>
           if (address != blockManager.blockManagerId) {
             shuffleMetrics.incRemoteBytesRead(buf.size)
             shuffleMetrics.incRemoteBlocksFetched(1)
           }
           buf.release()
+          freeMemory(size)
         case _ =>
       }
     }
+    shuffleFiles.foreach { shuffleFile =>
+      try {
+        shuffleFile.delete()
+      } catch {
+        case ioe: IOException =>
+          logError(s"Failed to cleanup ${shuffleFile.getAbsolutePath}.", ioe)
```

Comment:
`IOException` is not thrown by delete - but it can return `false` to indicate delete failure.
The log message (INFO would do btw) should be on `delete()` returning `false`.